pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -146,11 +146,52 @@ func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace st
 	glog.V(4).Infof("%s %s - sum of CPU requested: %d", namespace, selector, requestSum)
 	requestAvg := requestSum / int64(len(podList.Items))
 	// Consumption is already averaged and in millis.
-	consumption, timestamp, err := h.getForPods(heapsterCpuUsageMetricDefinition, namespace, podNames)
+	consumption, timestamp, err := h.getCpuUtilizationForPods(namespace, selector, podNames)
 	if err != nil {
 		return 0, 0, time.Time{}, err
 	}
-	return consumption.intValue, requestAvg, timestamp, nil
+	return consumption, requestAvg, timestamp, nil
+}
+
+func (h *HeapsterMetricsClient) getCpuUtilizationForPods(namespace string, selector labels.Selector, podNames map[string]bool) (int64, time.Time, error) {
+	metricPath := fmt.Sprintf("/apis/metrics/v1alpha1/namespaces/%s/pods", namespace)
+	params := map[string]string{"labelSelector": selector.String()}
+
+	resultRaw, err := h.client.Core().Services(h.heapsterNamespace).
+		ProxyGet(h.heapsterScheme, h.heapsterService, h.heapsterPort, metricPath, params).
+		DoRaw()
+	if err != nil {
+		return 0, time.Time{}, fmt.Errorf("failed to get pods metrics: %v", err)
+	}
+
+	glog.V(4).Infof("Heapster metrics result: %s", string(resultRaw))
+
+	metrics := make([]metrics_api.PodMetrics, 0)
+	err = json.Unmarshal(resultRaw, &metrics)
+	if err != nil {
+		return 0, time.Time{}, fmt.Errorf("failed to unmarshall heapster response: %v", err)
+	}
+
+	if len(metrics) != len(podNames) {
+		return 0, time.Time{}, fmt.Errorf("metrics obtained for %d/%d of pods", len(metrics), len(podNames))
+	}
+
+	sum := int64(0)
+	for _, m := range metrics {
+		if _, found := podNames[m.Name]; found {
+			for _, c := range m.Containers {
+				cpu, found := c.Usage[v1.ResourceCPU]
+				if !found {
+					return 0, time.Time{}, fmt.Errorf("no cpu for container %v in pod %v/%v", c.Name, namespace, m.Name)
+				}
+				sum += cpu.MilliValue()
+			}
+		} else {
+			return 0, time.Time{}, fmt.Errorf("not expected metrics for pod %v/%v", namespace, m.Name)
+		}
+	}
+
+	return sum / int64(len(metrics)), metrics[0].Timestamp.Time, nil
```

Comment:
As @kargakis no. Especially there is a test which verifies this `TestCPUEmptyMetrics`